### PR TITLE
Speed up workspace loading in the Muon Interface

### LINKED ISSUE #27590: Script generation without group workspace creating duplicates
If you open the muon analysis GUI and load some data, then get the history you will get a script that looks similar to the following:

```
LoadMuonNexus(Filename=r'\\musr\data\MUSR00062260.nxs', OutputWorkspace='MUSR00062260.nxs', DeadTimeTable='MUSR00062260.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
RenameWorkspace(InputWorkspace='MUSR00062260.nxs_deadtime_table', OutputWorkspace='MUSR62260_deadtime MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA', OutputWorkspace='MUSR62260 MA')
RenameWorkspace(InputWorkspace='MUSR00062260.nxs', OutputWorkspace='MUSR62260_raw_data MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA', OutputWorkspace='MUSR62260 MA')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; top; Counts; MA', GroupName='top', Grouping='17-24,49-56')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; top; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; top; Asymmetry; MA_unnorm', GroupName='top', Grouping='17-24,49-56', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bkwd; Counts; MA', GroupName='bkwd', Grouping='25-32,41-48')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bkwd; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; bkwd; Asymmetry; MA_unnorm', GroupName='bkwd', Grouping='25-32,41-48', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bottom; Counts; MA', GroupName='bottom', Grouping='1-8,33-40')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bottom; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; bottom; Asymmetry; MA_unnorm', GroupName='bottom', Grouping='1-8,33-40', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; fwd; Counts; MA', GroupName='fwd', Grouping='9-16,57-64')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; fwd; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; fwd; Asymmetry; MA_unnorm', GroupName='fwd', Grouping='9-16,57-64', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; bottom; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; bottom; Asymmetry; MA,MUSR62260; Group; fwd; Counts; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; bottom; Asymmetry; MA,MUSR62260; Group; fwd; Counts; MA,MUSR62260; Group; fwd; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
```

after a little reordering (this wont change the results):
```
LoadMuonNexus(Filename=r'\\musr\data\MUSR00062260.nxs', OutputWorkspace='MUSR00062260.nxs', DeadTimeTable='MUSR00062260.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
RenameWorkspace(InputWorkspace='MUSR00062260.nxs_deadtime_table', OutputWorkspace='MUSR62260_deadtime MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA', OutputWorkspace='MUSR62260 MA')
RenameWorkspace(InputWorkspace='MUSR00062260.nxs', OutputWorkspace='MUSR62260_raw_data MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA', OutputWorkspace='MUSR62260 MA')


MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')


MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; top; Counts; MA', GroupName='top', Grouping='17-24,49-56')
MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bkwd; Counts; MA', GroupName='bkwd', Grouping='25-32,41-48')
MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bottom; Counts; MA', GroupName='bottom', Grouping='1-8,33-40')
MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; fwd; Counts; MA', GroupName='fwd', Grouping='9-16,57-64')


MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; top; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; top; Asymmetry; MA_unnorm', GroupName='top', Grouping='17-24,49-56', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bkwd; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; bkwd; Asymmetry; MA_unnorm', GroupName='bkwd', Grouping='25-32,41-48', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bottom; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; bottom; Asymmetry; MA_unnorm', GroupName='bottom', Grouping='1-8,33-40', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; fwd; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; fwd; Asymmetry; MA_unnorm', GroupName='fwd', Grouping='9-16,57-64', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)


GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; bottom; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; bottom; Asymmetry; MA,MUSR62260; Group; fwd; Counts; MA', OutputWorkspace='MUSR62260 MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; bottom; Asymmetry; MA,MUSR62260; Group; fwd; Counts; MA,MUSR62260; Group; fwd; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
```
It becomes apparent that we only need to do the below line once:
```
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
```
but its repeated multiple times.

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
Issues with loading in the Muon interface have been noted previously, where it was found that the workspace history, for a run loaded in the GUI, appears to be quite bloated. 

In this PR it was found that the MuonPreProcess and GroupWorkspaces algorithm were being run for each detector group, which feels quite unnecessary. This PR minimises the number of calls made to these algorithms, to improve the speed of loading and to make the workspace history appear more sensible.

This PR implements a context manager to stall the execution of the grouping algorithm. 

Comparing the workspace history before for MUSR62260 loaded in the GUI (**28 Algorithm calls**):

```
  LoadMuonNexus(Filename=r'\\isis.cclrc.ac.uk\inst$\ndxmusr\instrument\data\cycle_16_5\MUSR00062260.nxs', OutputWorkspace='MUSR00062260.nxs', DeadTimeTable='MUSR00062260.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
    RenameWorkspace(InputWorkspace='MUSR00062260.nxs_deadtime_table', OutputWorkspace='MUSR62260_deadtime MA')
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA', OutputWorkspace='MUSR62260 MA')
    RenameWorkspace(InputWorkspace='MUSR00062260.nxs', OutputWorkspace='MUSR62260_raw_data MA')
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA', OutputWorkspace='MUSR62260 MA')
    MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
    MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; top; Counts; MA', GroupName='top', Grouping='17-24,49-56')
    MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
    MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; top; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; top; Asymmetry; MA_unnorm', GroupName='top', Grouping='17-24,49-56', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
    MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
    MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bkwd; Counts; MA', GroupName='bkwd', Grouping='25-32,41-48')
    MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
    MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bkwd; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; bkwd; Asymmetry; MA_unnorm', GroupName='bkwd', Grouping='25-32,41-48', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
    MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
    MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bottom; Counts; MA', GroupName='bottom', Grouping='1-8,33-40')
    MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
    MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; bottom; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; bottom; Asymmetry; MA_unnorm', GroupName='bottom', Grouping='1-8,33-40', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
    MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
    MuonGroupingCounts(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; fwd; Counts; MA', GroupName='fwd', Grouping='9-16,57-64')
    MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
    MuonGroupingAsymmetry(InputWorkspace='__pre_processed_data', OutputWorkspace='MUSR62260; Group; fwd; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; fwd; Asymmetry; MA_unnorm', GroupName='fwd', Grouping='9-16,57-64', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA', OutputWorkspace='MUSR62260 MA')
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA', OutputWorkspace='MUSR62260 MA')
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA', OutputWorkspace='MUSR62260 MA')
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; bottom; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; bottom; Asymmetry; MA,MUSR62260; Group; fwd; Counts; MA', OutputWorkspace='MUSR62260 MA')
    GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA,MUSR62260; Group; top; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; bottom; Asymmetry; MA,MUSR62260; Group; fwd; Counts; MA,MUSR62260; Group; fwd; Asymmetry; MA', OutputWorkspace='MUSR62260 MA')

```

And now (**14 Algorithm calls**)
```
LoadMuonNexus(Filename=r'\\isis.cclrc.ac.uk\inst$\ndxmusr\instrument\data\cycle_16_5\MUSR00062260.nxs', OutputWorkspace='MUSR00062260.nxs', DeadTimeTable='MUSR00062260.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
RenameWorkspace(InputWorkspace='MUSR00062260.nxs_deadtime_table', OutputWorkspace='MUSR62260_deadtime MA')
RenameWorkspace(InputWorkspace='MUSR00062260.nxs', OutputWorkspace='MUSR62260_raw_data MA')
GroupWorkspaces(InputWorkspaces='MUSR62260_deadtime MA,MUSR62260_raw_data MA', OutputWorkspace='MUSR62260 MA')
MuonPreProcess(InputWorkspace='MUSR62260_raw_data MA', OutputWorkspace='__MUSR62260_pre_processed_data', TimeMin=0.10000000000000001, TimeOffset=0, DeadTimeTable='MUSR62260_deadtime MA')
MuonGroupingCounts(InputWorkspace='__MUSR62260_pre_processed_data', OutputWorkspace='MUSR62260; Group; top; Counts; MA', GroupName='top', Grouping='17-24,49-56')
MuonGroupingAsymmetry(InputWorkspace='__MUSR62260_pre_processed_data', OutputWorkspace='MUSR62260; Group; top; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; top; Asymmetry; MA_unnorm', GroupName='top', Grouping='17-24,49-56', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
MuonGroupingCounts(InputWorkspace='__MUSR62260_pre_processed_data', OutputWorkspace='MUSR62260; Group; bkwd; Counts; MA', GroupName='bkwd', Grouping='25-32,41-48')
MuonGroupingAsymmetry(InputWorkspace='__MUSR62260_pre_processed_data', OutputWorkspace='MUSR62260; Group; bkwd; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; bkwd; Asymmetry; MA_unnorm', GroupName='bkwd', Grouping='25-32,41-48', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
MuonGroupingCounts(InputWorkspace='__MUSR62260_pre_processed_data', OutputWorkspace='MUSR62260; Group; bottom; Counts; MA', GroupName='bottom', Grouping='1-8,33-40')
MuonGroupingAsymmetry(InputWorkspace='__MUSR62260_pre_processed_data', OutputWorkspace='MUSR62260; Group; bottom; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; bottom; Asymmetry; MA_unnorm', GroupName='bottom', Grouping='1-8,33-40', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
MuonGroupingCounts(InputWorkspace='__MUSR62260_pre_processed_data', OutputWorkspace='MUSR62260; Group; fwd; Counts; MA', GroupName='fwd', Grouping='9-16,57-64')
MuonGroupingAsymmetry(InputWorkspace='__MUSR62260_pre_processed_data', OutputWorkspace='MUSR62260; Group; fwd; Asymmetry; MA', OutputUnNormWorkspace='__MUSR62260; Group; fwd; Asymmetry; MA_unnorm', GroupName='fwd', Grouping='9-16,57-64', AsymmetryTimeMin=0.10000000000000001, AsymmetryTimeMax=32.293998718261719)
GroupWorkspaces(InputWorkspaces='MUSR62260; Group; fwd; Counts; MA,MUSR62260_raw_data MA,MUSR62260; Group; bottom; Counts; MA,MUSR62260; Group; fwd; Asymmetry; MA,MUSR62260; Group; bottom; Asymmetry; MA,MUSR62260; Group; bkwd; Asymmetry; MA,MUSR62260; Group; bkwd; Counts; MA,MUSR62260; Group; top; Asymmetry; MA,MUSR62260_deadtime MA,MUSR62260; Group; top; Counts; MA', OutputWorkspace='MUSR62260 MA')

```

Resulted in a speed up from 1.2s to 0.7s on my Workbench Debug build for loading MUSR62260 in the interface. 

**To test:**
From my observations removing these algorithm calls has not affected the data that is loaded.

To test you can:
1. Using a master build load data in the Muon Analysis 2 GUI
2. Save the workspace history as a script (**script A**)
3. Switch to this branch
4. Load the same data in the Muon Analysis 2 GUI
5. Save the workspace history as a script (**script B**)
6. Change the Workspace group name in **script B**
7. Compare the workspaces created by **script A** and **script B,** they should be the same.

Fixes #27590. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Harrietbrown — CHANGES_REQUESTED** [review]
one question

File: scripts/Muon/GUI/Common/ADSHandler/muon_workspace_wrapper.py [hunk 1]
```diff
@@ -118,6 +120,7 @@ def show(self, name=''):
         if len(self.name) > 0:
             # add workspace to ADS
             if self.is_hidden:
+                print(self.name, self._workspace_name, self.workspace)
```
[the remark this answers is not in the document]
Comment (StephenSmith25) [reply scripts/Muon/GUI/Common/ADSHandler/muon_workspace_wrapper.py:123]: Nope, completly forgot I put that there 🤦

**Review by Harrietbrown — DISMISSED** [review]
Very clever soluton on the group workspace I expect ther may be other places where that helper will be useful. In this I got a significant speed up on load time, 8.4 seconds without the PR, 4.82s with the PR, so a good job all around.

**Comment by gvardany** [comment]
Is this worth a release note?

**Review by Harrietbrown — APPROVED** [review]
Release notes look ok.

**Pull request merged**